presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/csv/CsvColumnEncoder.java

```diff
@@ -0,0 +1,116 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.csv;
+
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.plugin.kafka.encoder.EncoderColumnHandle;
+import io.prestosql.plugin.kafka.encoder.EncoderFieldValueProvider;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.type.Type;
+
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.plugin.kafka.encoder.EncoderErrorCode.ENCODER_CONVERSION_NOT_SUPPORTED;
+import static io.prestosql.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.spi.type.RealType.REAL;
+import static io.prestosql.spi.type.SmallintType.SMALLINT;
+import static io.prestosql.spi.type.TinyintType.TINYINT;
+import static io.prestosql.spi.type.Varchars.isVarcharType;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class CsvColumnEncoder
+{
+    private static final Set<Type> SUPPORTED_PRIMITIVE_TYPES = ImmutableSet.of(
+            BOOLEAN,
+            TINYINT,
+            SMALLINT,
+            INTEGER,
+            BIGINT,
+            DOUBLE,
+            REAL);
+
+    private final String columnName;
+    private final Type columnType;
+
+    public CsvColumnEncoder(EncoderColumnHandle columnHandle)
+    {
+        try {
+            requireNonNull(columnHandle, "columnHandle is null");
+            this.columnName = columnHandle.getName();
+            this.columnType = columnHandle.getType();
+
+            checkArgument(!columnHandle.isInternal(), "unexpected internal column'%s'", columnName);
+            checkArgument(columnHandle.getFormatHint() == null, "unexpected format hint '%s' defined for column '%s'", columnHandle.getFormatHint(), columnName);
+            checkArgument(columnHandle.getDataFormat() == null, "unexpected data format '%s' defined for column '%s'", columnHandle.getDataFormat(), columnName);
+
+            checkArgument(isSupportedType(columnType), "unsupported column type '%s' for column '%s'", columnType, columnName);
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(GENERIC_USER_ERROR, e);
+        }
+    }
+
+    private boolean isSupportedType(Type type)
```

Comment:
This method is unnecessary.  It always produces the same result as `isSupportedPrimitive()`.  Rename the other method to `isSupportedType` and drop this one.

Reply:
I had that there in case we add complex types, but I guess that is unlikely for csv format